spec/components/topic_view_spec.rb

```diff
@@ -249,6 +249,20 @@
       end
     end
 
+    describe "#get_canonical_path with embed_set_canonical_url true" do
+      fab!(:topic_embed) { Fabricate(:topic_embed, embed_url: "https://markvanlan.com") }
+      let(:path) { "/1234" }
+
+      before do
+        SiteSetting.embed_set_canonical_url = true
+        TopicView.any_instance.stubs(:find_topic).with(1234).returns(topic_embed.topic)
```

Comment:
Could you do this without a stub? I'd always prefer that.

It might be easier to do a request spec for this particular thing.

Reply:
@eviltrout Agreed; moved to topics_controller_spec without stubbing. Also this now tests all the way from server to crawler specifically which rules. Great call.